apps/autorelayer-interop/src/relayer.ts

```diff
@@ -330,26 +346,63 @@ export class Relayer {
     return { client, walletClient, account }
   }
 
+  private getGasProviderBalance(gasProvider: GasTankProvider) {
+    if (gasProvider.pendingWithdrawal) {
+      const balance =
+        gasProvider.gasProviderBalance - gasProvider.pendingWithdrawal.amount
```

Comment:
these could also be replaced with rpc calls to the `GasTank` contract in order to get latest balance and withdrawals